flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisConsumer.java

```diff
@@ -0,0 +1,481 @@
+				default:
+					sentinelSequenceNum = SentinelSequenceNumber.SENTINEL_LATEST_SEQUENCE_NUM;
+			}
+
+			for (KinesisStreamShard assignedShard : assignedShards) {
+				fetcher.advanceSequenceNumberTo(assignedShard, sentinelSequenceNum.toString());
+			}
+
+			if (LOG.isInfoEnabled()) {
+				StringBuilder sb = new StringBuilder();
+				for (KinesisStreamShard assignedShard : assignedShards) {
+					sb.append(assignedShard.getStreamName()).append(":").append(assignedShard.getShardId())
+						.append(" -> ").append(sentinelSequenceNum.toString()).append(", ");
+				}
+				LOG.info("Advanced the starting sequence numbers of consumer task {}: {}", thisConsumerTaskIndex, sb.toString());
+			}
+		}
+	}
+
+	@Override
+	public void run(SourceContext<T> sourceContext) throws Exception {
+		if (hasAssignedShards) {
+			fetcher.run(sourceContext, deserializer, lastSequenceNums);
+		} else {
+			// this source never completes because there is no assigned shards,
+			// so emit a Long.MAX_VALUE watermark to no block watermark forwarding
+			sourceContext.emitWatermark(new Watermark(Long.MAX_VALUE));
+
+			final Object waitLock = new Object();
+			while (running) {
+				try {
+					synchronized (waitLock) {
+						waitLock.wait();
+					}
+				} catch (InterruptedException e) {
+					// do nothing
+				}
+			}
+		}
+
+		sourceContext.close();
+	}
+
+	@Override
+	public void cancel() {
+		running = false;
+
+		// interrupt the fetcher of any work
+		KinesisDataFetcher fetcher = this.fetcher;
+		this.fetcher = null;
+		if (fetcher != null) {
+			try {
+				fetcher.close();
+			} catch (IOException e) {
+				LOG.warn("Error while closing Kinesis data fetcher", e);
+			}
+		}
+	}
+
+	@Override
+	public void close() throws Exception {
+		cancel();
+		super.close();
+	}
+
+	@Override
+	public TypeInformation<T> getProducedType() {
+		return deserializer.getProducedType();
+	}
+
+	// ------------------------------------------------------------------------
+	//  Checkpoint and restore
+	// ------------------------------------------------------------------------
+
+	@Override
+	public HashMap<KinesisStreamShard, String> snapshotState(long checkpointId, long checkpointTimestamp) throws Exception {
+		if (lastSequenceNums == null) {
+			LOG.debug("snapshotState() requested on not yet opened source; returning null.");
+			return null;
+		}
+
+		if (!running) {
+			LOG.debug("snapshotState() called on closed source; returning null.");
+			return null;
+		}
+
+		if (LOG.isDebugEnabled()) {
+			LOG.debug("Snapshotting state. ...");
+		}
+
+		@SuppressWarnings("unchecked")
+		HashMap<KinesisStreamShard, String> currentSequenceNums = (HashMap<KinesisStreamShard, String>) lastSequenceNums.clone();
+		return currentSequenceNums;
+	}
+
+	@Override
+	public void restoreState(HashMap<KinesisStreamShard, String> restoredState) throws Exception {
+		sequenceNumsToRestore = restoredState;
+	}
+
+	@Override
+	public void notifyCheckpointComplete(long checkpointId) throws Exception {
```

Comment:
This method can be removed (along with its interface).
Its just logging debug info's logged by the `Task` as well.

In the future, if we add support for committing into Dynamodb or Zookeeper, we might re-introduce the method.